src/adapt.jl

```diff
@@ -32,7 +35,8 @@ function do_quadgk(f::F, s::NTuple{N,T}, n, atol, rtol, maxevals, nrm) where {T,
         return (I, E) # fast return when no subdivisions required
```

Comment:
A basic issue here is that it's nice to support allocation-free quadrature in cases when no subdivisions are required, which is the case when tuples are used as above.